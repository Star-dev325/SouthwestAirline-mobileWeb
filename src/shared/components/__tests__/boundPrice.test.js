import { render } from '@testing-library/react';
import React from 'react';
import BoundPrice from 'src/shared/components/boundPrice';

describe('BoundPrice', () => {
  const defaultProps = {
    arrivalAirportCode: 'HOU',
    departureAirportCode: 'DAL',
    flight: '4L9D73',
    paxCount: 2,
    price: {
      amount: '40.00',
      currencyCode: 'USD',
      currencySymbol: '$'
    },
    priceString: null,
    totalPrice: {
      amount: '80.00',
      currencyCode: 'USD',
      currencySymbol: '$'
    }
  };

  it('should render correctly', () => {
    const { container } = render(<BoundPrice {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should render a priceString if one is provided', () => {
    const props = {
      ...defaultProps,
      priceString: '#mock-price-string'
    };

    const { container } = render(<BoundPrice {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('it should render points correctly', () => {
    const props = {
      ...defaultProps,
      price: {
        amount: '2,000',
        currencyCode: 'PTS',
        currencySymbol: null
      },
      totalPrice: {
        amount: '4,000',
        currencyCode: 'PTS',
        currencySymbol: null
      }
    };

    const { container } = render(<BoundPrice {...props} />);

    expect(container).toMatchSnapshot();
  });
});
