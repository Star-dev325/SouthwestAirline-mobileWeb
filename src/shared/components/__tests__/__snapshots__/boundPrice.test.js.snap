// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoundPrice it should render points correctly 1`] = `
<div>
  <div
    class="bound-price"
  >
    <div
      class="bound-price--airport-section"
    >
      <div
        class="bound-price--airport-info"
      >
        DAL
         - 
        HOU
      </div>
      <div
        class="bound-price--flight-number"
      >
        4L9D73
      </div>
    </div>
    <div
      class="bound-price--amount-info-section"
    >
      <div
        class="bound-price--amount-section"
      >
        <div
          class="bound-price--amount-flex-group"
        >
          <div
            class="bound-price--amount"
          >
            <div
              class="currency"
            >
              <span
                data-qa="total-amount"
              >
                2,000
              </span>
            </div>
          </div>
          <div
            class="bound-price--pax-count"
          >
            x
            2
             
            AIR_UPGRADE_MULTI_PAX_TEXT
          </div>
        </div>
        <div
          class="bound-price--total-amount"
        >
          <div
            class="currency"
          >
            <span
              data-qa="total-amount"
            >
              4,000
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BoundPrice should render a priceString if one is provided 1`] = `
<div>
  <div
    class="bound-price"
  >
    <div
      class="bound-price--airport-section"
    >
      <div
        class="bound-price--airport-info"
      >
        DAL
         - 
        HOU
      </div>
      <div
        class="bound-price--flight-number"
      >
        4L9D73
      </div>
    </div>
    <div
      class="bound-price--amount-info-section"
    >
      <div
        class="bound-price--amount-section"
      >
        <div
          class="bound-price--amount-flex-group"
        >
          <div
            class="bound-price--amount"
          >
            #mock-price-string
          </div>
          <div
            class="bound-price--pax-count"
          >
            x
            2
             
            AIR_UPGRADE_MULTI_PAX_TEXT
          </div>
        </div>
        <div
          class="bound-price--total-amount"
        >
          <div
            class="currency"
          >
            <span
              class="money-sign"
            >
              $
            </span>
            <span
              data-qa="total-amount"
            >
              80.00
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BoundPrice should render correctly 1`] = `
<div>
  <div
    class="bound-price"
  >
    <div
      class="bound-price--airport-section"
    >
      <div
        class="bound-price--airport-info"
      >
        DAL
         - 
        HOU
      </div>
      <div
        class="bound-price--flight-number"
      >
        4L9D73
      </div>
    </div>
    <div
      class="bound-price--amount-info-section"
    >
      <div
        class="bound-price--amount-section"
      >
        <div
          class="bound-price--amount-flex-group"
        >
          <div
            class="bound-price--amount"
          >
            <div
              class="currency"
            >
              <span
                class="money-sign"
              >
                $
              </span>
              <span
                data-qa="total-amount"
              >
                40.00
              </span>
            </div>
          </div>
          <div
            class="bound-price--pax-count"
          >
            x
            2
             
            AIR_UPGRADE_MULTI_PAX_TEXT
          </div>
        </div>
        <div
          class="bound-price--total-amount"
        >
          <div
            class="currency"
          >
            <span
              class="money-sign"
            >
              $
            </span>
            <span
              data-qa="total-amount"
            >
              80.00
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
